frontend/test/lit/test_function_attributes.py

```diff
@@ -28,7 +28,7 @@ def qnode(x):
 
 @qjit(target="mlir")
 # The entry point has no internal linkage.
-# CHECK-DAG: func.func public @jit.workload(%arg0: tensor<f64>) -> tensor<4xcomplex<f64>> attributes {llvm.emit_c_interface} {
+# CHECK-DAG: func.func public @jit_workload(%arg0: tensor<f64>) -> tensor<4xcomplex<f64>> attributes {llvm.emit_c_interface, llvm.linkage = #llvm.linkage<internal>} {
```

Comment:
This one is actually an interesting issue. I believe that `jit_workload` should still not have internal linkage. [Internal linkage ](https://llvm.org/docs/LangRef.html#linkage-types) corresponds to the `static` keyword in C. I am surprised that I was able to link against it in the wrapper.

Reply:
So the wrapper and our module are compiled in separate compilation units? That is odd that it was still able to link then 🤔 Do we not control this linkage flag (since you think it shouldn't be there)?

Reply:
We do control this linkage flag. Do remember that without it we get the error that if a function has the same name as another function already loaded by the Procedure Linkage Table (PLT), then the one that was loaded by the PLT will be the target. It might be that it was statically linked and since we generate llvm bytecode but not C we might have missed another attribute to match C's `static` annotation semantics? Or a bug in the linker. Or a bug in our assumptions of this `internal` linkage attribute. I think I'll investigate this when I have some cycles to spare, but leave it be for now.

Reply:
Sounds good

Reply:
What is the point of that comment in this test, as in some new developer reading it now what does it tell them?

Reply:
This test explicitly checks that the entry point to the translation unit of the quantum program has no internal linkage while others (see qnode) have internal linkage. Sure. I can delete the comment.

Reply:
Oh got it, you removed the internal linkage in https://github.com/PennyLaneAI/catalyst/pull/97/commits/f3fd03ebdb1d4e769e1cc47830f194ebe085b775